### DISCUSSION TITLE
Multiple-vehicles support in Airsim using Gazebo drone project

### DISCUSSION THREAD

**Discussion by airsimdevdd** [microsoft/AirSim#4246] [General]
Any one could help me with multiple vehicle in airsim using gazebo drone  for fixed wing sitl what are changes that should I change in settings.json file and gazebodrone  file  @JaeyoungLim

**Comment by zimmy87** [comment]
You'll need to update this [section of GazeboDrone](https://github.com/microsoft/AirSim/blob/master/GazeboDrone/src/main.cpp#L70-L75). Right now, it's hardcoded to only call `client.simSetVehiclePose` once during playback for the first pose in the ConstPosesStampedPtr parameter. This section needs to be changed so that `client.simSetVehiclePose` is called for each pose that corresponds to a vehicle and with the right vehicle_name parameter. You can set vehicle names in the [vehicle settings](https://microsoft.github.io/AirSim/settings/#vehicles-settings).
